indico/modules/events/registration/client/js/components/ParticipantListAccordion.tsx

```diff
@@ -0,0 +1,238 @@
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {HTMLAttributes, ReactNode, useState} from 'react';
+import {
+  AccordionTitle,
+  AccordionContent,
+  Accordion,
+  Icon,
+  TableRow,
+  TableHeaderCell,
+  TableHeader,
+  TableCell,
+  TableBody,
+  Table,
+  TableFooter,
+  Popup,
+  Message,
+} from 'semantic-ui-react';
+
+import {Param, Plural, PluralTranslate, Singular, Translate} from 'indico/react/i18n';
+
+import './ParticipantListAccordion.module.scss';
+
+interface TableObj {
+  headers: string[];
+  rows: [
+    {
+      checked_in: boolean;
+      columns: [{text: string; is_picture: boolean; sort_key: string | null}];
+    }
+  ];
+  num_participants: number;
+  show_checkin: boolean;
+  title?: string;
+}
+
+interface ParticipantListAccordionProps {
+  tables: TableObj[];
+}
+
+interface AccordionParticipantsItemProps {
+  index: number;
+  table: TableObj;
+  children?: ReactNode;
+  collapsible?: boolean;
+}
+
+interface ParticipantCounterProps extends HTMLAttributes<HTMLSpanElement> {
+  totalCount: number;
+  hiddenCount: number;
+  styleName?: string;
+}
+
+const ParticipantCountTranslationHidden: React.FC<{count: number}> = ({count}) =>
+  count > 0 ? (
+    <PluralTranslate count={count}>
+      <Singular>
+        <Param name="anonymous-participants-count" value={count} /> participant registered
+        anonymously.
+      </Singular>
+      <Plural>
+        <Param name="anonymous-participants-count" value={count} /> participants registered
+        anonymously.
+      </Plural>
+    </PluralTranslate>
+  ) : (
+    <Translate>No anonymous participants registered</Translate>
+  );
+
+const ParticipantCounter: React.FC<ParticipantCounterProps> = ({
+  totalCount,
+  hiddenCount,
+  styleName = '',
+  ...props
+}) => (
+  <div className={styleName} {...props}>
+    <Popup
+      position="left center"
+      content={<ParticipantCountTranslationHidden count={hiddenCount} />}
+      trigger={
+        <span>
+          {totalCount} <Icon name="user" />
+        </span>
+      }
+    />
+  </div>
+);
+
+function ParticipantTable({table}: {table: TableObj}) {
+  const visibleParticipantsCount = table.rows.length;
+  const totalParticipantCount = table.num_participants;
+  const hiddenParticipantsCount = totalParticipantCount - visibleParticipantsCount;
+  const hasInvisibleParticipants = hiddenParticipantsCount > 0;
+
+  type sortDirectionType = 'ascending' | 'descending' | null;
+
+  const [sortColumn, setSortColumn] = useState<number | null>(null);
+  const [sortDirection, setSortDirection] = useState<sortDirectionType>(null);
+  const [sortedRows, setSortedRows] = useState([...table.rows]);
+
+  const handleSort = (columnIndex: number) => {
+    const currentSortColumn = sortColumn === columnIndex ? sortColumn : columnIndex;
+
+    const directions: Record<sortDirectionType, sortDirectionType> = {
+      [null as symbol]: 'ascending',
```

Comment:
I'd use the type of the Record
```suggestion
      [null as  as sortDirectionType]: 'ascending',
```

Reply:
Ah yes of course! Thanks